File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/roboti/activity/LevelRaceActivity.java
package cz.muni.fi.umimecesky.roboti.activity;


import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import java.util.List;

import cz.muni.fi.umimecesky.roboti.R;
import cz.muni.fi.umimecesky.roboti.adapterlistener.LevelAdapter;
import cz.muni.fi.umimecesky.roboti.pojo.RaceConcept;
import cz.muni.fi.umimecesky.roboti.utils.Constant;
import cz.muni.fi.umimecesky.roboti.utils.Utils;

import static cz.muni.fi.umimecesky.roboti.utils.Utils.getWebConcepts;

public class LevelRaceActivity extends AppCompatActivity {

    private LevelAdapter levelAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_race_levels);
    }

    @Override
    protected void onStart() {
        // TODO: replace with data update - notify?
        super.onStart();
        ListView listView = (ListView) findViewById(R.id.levelListView);

        levelAdapter = new LevelAdapter(this, getWebConcepts(this));
        listView.setAdapter(levelAdapter);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, final View view, int position, long id) {
                List<RaceConcept> conceptList = Utils.getWebConcepts(LevelRaceActivity.this);
                RaceConcept raceConcept = conceptList.get(position);
                Log.v("race concept", String.valueOf(raceConcept));

                Intent intent = new Intent(LevelRaceActivity.this, RaceActivity.class);
                intent.putExtra(Constant.RACE_CONCEPT_EXTRA, raceConcept);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/cz/muni/fi/umimecesky/game/robots/LevelRaceActivity.kt
/*
 * Copyright (c) 2018 Marek Sabo
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *                http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package cz.muni.fi.umimecesky.game.robots


import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import cz.muni.fi.umimecesky.R
import cz.muni.fi.umimecesky.game.shared.util.GuiUtil.hideNavigationBar
import cz.muni.fi.umimecesky.prefs
import kotlinx.android.synthetic.main.activity_race_levels.levelListView
import org.jetbrains.anko.startActivity

class LevelRaceActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_race_levels)

    }

    override fun onStart() {
        // TODO: replace with data update - notify?
        super.onStart()

        levelListView.layoutManager = LinearLayoutManager(this)
        levelListView.itemAnimator = DefaultItemAnimator()

        levelListView.adapter =
                LevelAdapter(prefs.maxRobotsCategories, { raceConcept ->
                    prefs.currentRobotConcept = raceConcept
                    startActivity<RaceActivity>()
                })

    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) hideNavigationBar()
    }

}
